misc/sample_application/Makefile

```diff
@@ -0,0 +1,18 @@
+T := $(CURDIR)
+OUT_DIR ?= $(shell mkdir -p $(T)/build;cd $(T)/build/;pwd)
+
+.PHONY: all userapp rtapp
+all: userapp rtapp
+
+userapp:
+	cp $(T)/uservm/histapp.py $(OUT_DIR)
+	$(MAKE) -C $(T)/uservm  -o=$(OUT_DIR)
```

Comment:
Do you want to pass `OUT_DIR` to the sub-makefiles? `-o` is not for that purpose.

You should use `OUT_DIR=$(OUT_DIR)` here and use `$(OUT_DIR)` rather than hardcoding `../build` in the sub-makefiles.

Reply:
Re-push the new modification about the OUT_DIR.

Reply:
You are still not using `$(OUT_DIR)` in the sub-makefiles, which means passing `OUT_DIR` to them has no effect at all.

Reply:
1 . Modify the Makefile in the parent directory pass the OUT_DIR to the sub-makefies.
2. Modify the Makefile in the sub-directory in case the user just run the make command in the sub-directory.
Fore update the commit in this PR